src/ui/events.js

```diff
@@ -10,29 +10,28 @@ import type Map from './map.js';
 import type LngLat from '../geo/lng_lat.js';
 
 /**
- * `MapMouseEvent` is the event type for mouse-related map events.
+ * `MapMouseEvent` is the type for objects emitted by mouse-related [`Map` events](https://docs.mapbox.com/mapbox-gl-js/api/map/#map-events).
  * @extends {Object}
  * @example
- * // The `click` event is an example of a `MapMouseEvent`.
- * // Set up an event listener on the map.
- * map.on('click', function(e) {
- *   // The event object (e) contains information like the
- *   // coordinates of the point on the map that was clicked.
- *   console.log('A click event has occurred at ' + e.lngLat);
- * });
+ * // Example of a MapMouseEvent of type "click"
+ * {
+ *     lngLat: {
+ *         lng: 40.203,
+ *         lat: -74.451
+ *     },
+ *     originalEvent: {...},
+ *     point: {
+ *         x: 266,
+ *         y: 464
+ *     },
+ *      preventDefault(),
```

Comment:
I also wasn't super sure how to handle `preventDefault`.

Reply:
I agree. When I actually looked at the `e` object for a different type (`MapWheelEvent`) it looked like this: `_defaultPrevented: false` which seems quite awkward...but that is in fact what I think a user should expect from an object of this type! 🤷🏼

Reply:
@mapbox/gl-js team, should we omit methods like `preventDefault` from these examples?

Reply:
I think so. It's awkward because you can call `e.preventDefault()` but if you log the `e` object, you won't see the method directly (it's in the [prototype chain](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Inheritance_and_the_prototype_chain) instead). The issue here is that these map events are designed to work the same way that DOM browser events do, but they're not actually DOM browser events. Since the entire GL JS map is a single DOM element, adding map events requires us to create those events internally in GL JS. It also means that preventing the default action of an element won't prevent it bubbling up since the DOM isn't controlling them; this is why `preventDefault` is implemented this way.